vite.config.ts

```diff
@@ -31,23 +30,27 @@ const serverSettings = {
     },
 };
 
-export default defineConfig({
+export default defineConfig((config) => ({
     plugins: [
         react(),
         eslint({
-            failOnWarning: true,
+            failOnWarning: config.mode !== 'development',
+            lintOnStart: config.command === 'serve',
         }),
         svgr(), // works on every import with the pattern "**/*.svg?react"
         reactVirtualized(),
         tsconfigPaths(), // to resolve absolute path via tsconfig cf https://stackoverflow.com/a/68250175/5092999
     ],
+    resolve: {
+        preserveSymlinks: true, //else eslint don't detect "npm link" packages
```

Comment:
Why would we want that ? Which case was a problem ?

Reply:
when using `npm link` (to test commons-ui for example), we Vite call eslint with symlinks resolved and analyse the node_modules of the linked dependencies because it is outside of the node_modules of the project

Reply:
Ok got it thanks